Release of 5.0.0

dmatej opened the pull request:
Recreated PR of https://github.com/eclipse-ee4j/orb/pull/221 which was closed by GitHub after restaging.

pzygielo commented:
@dmatej It seems that release job didn't push updated tag. The current 5.0.0 is attached to 7ea6269, but the re-release created 43c7d51. Can you check the job if it forces push or deletes pre-existing tag in case of rebuild, please?
(Luckily `git diff 43c7d51 7ea6269` is empty, but in the future it might not be the case.)

dmatej commented:
> @dmatej It seems that release job didn't push updated tag. The current 5.0.0 is attached to [7ea6269](https://github.com/eclipse-ee4j/orb/commit/7ea6269ddf7ce3bbb6d0f549a38ec69789497df1), but the re-release created [43c7d51](https://github.com/eclipse-ee4j/orb/commit/43c7d5126f71fcf8c752c5e988a143f2cb68577d). Can you check the job if it forces push or deletes pre-existing tag in case of rebuild, please? (Luckily `git diff 43c7d51 7ea6269` is empty, but in the future it might not be the case.)
> 
> Similarly
> 
> * [Release of 4.1.0 orb-gmbal#59](https://github.com/eclipse-ee4j/orb-gmbal/pull/59)

Good you noticed that. I have fixed it here, yet the other one repository. I will check the job too.

dmatej commented:
```
++ git ls-remote --tags origin
++ grep -o 'refs/tags/5.0.0$'
+ [[ -n refs/tags/5.0.0 ]]
+ [[ true = \t\r\u\e ]]
+ echo '5.0.0 tag already exists, deleting'
5.0.0 tag already exists, deleting
+ git push --delete origin 5.0.0
To github.com:eclipse-ee4j/orb.git
 - [deleted]         5.0.0
+ true
+ git branch --delete 5.0.0-BRANCH
error: branch '5.0.0-BRANCH' not found
+ git checkout -b 5.0.0-BRANCH
Switched to a new branch '5.0.0-BRANCH'
+ git tag --delete 5.0.0
Deleted tag '5.0.0' (was efd2cce)
```
I see this in the log file, so I guess it worked, but probably the cause could be that I am often using "rebase and merge", which would change the commit id. However I cannot confirm that. However it seems the "Prepare next development cycle" commit changed the id too, so it could be this cause. I don't have another explanation ...

pzygielo commented:
Ok, good to know that job is fine.

> rebase and merge", which would change the commit id

Indeed, 778ba05 which was the head of this PR is not linked to master now. Its parent 7ea6269 was the one tagged, and on merge it was rewritten as 43c7d51.
